crates/nu-plugin-core/src/interface/mod.rs

```diff
@@ -185,31 +180,10 @@ pub trait InterfaceManager {
                 let reader = handle.read_stream(info.id, self.get_interface())?;
                 ListStream::new(reader, info.span, ctrlc.cloned()).into()
             }
-            PipelineDataHeader::ExternalStream(info) => {
+            PipelineDataHeader::ByteStream(info) => {
                 let handle = self.stream_manager().get_handle();
-                let span = info.span;
-                let new_raw_stream = |raw_info: RawStreamInfo| {
-                    let reader = handle.read_stream(raw_info.id, self.get_interface())?;
-                    let mut stream =
-                        RawStream::new(Box::new(reader), ctrlc.cloned(), span, raw_info.known_size);
-                    stream.is_binary = raw_info.is_binary;
-                    Ok::<_, ShellError>(stream)
-                };
-                PipelineData::ExternalStream {
-                    stdout: info.stdout.map(new_raw_stream).transpose()?,
-                    stderr: info.stderr.map(new_raw_stream).transpose()?,
-                    exit_code: info
-                        .exit_code
-                        .map(|list_info| {
-                            handle
-                                .read_stream(list_info.id, self.get_interface())
-                                .map(|reader| ListStream::new(reader, info.span, ctrlc.cloned()))
-                        })
-                        .transpose()?,
-                    span: info.span,
-                    metadata: None,
-                    trim_end_newline: info.trim_end_newline,
-                }
+                let reader = handle.read_stream(info.id, self.get_interface())?;
+                ByteStream::from_result_iter(reader, info.span, None).into()
```

Comment:
In this case, `from_result_iter()` does need to take `ctrlc`, because the underlying stream is not checking it.